src/test/java/com/fasterxml/jackson/databind/ObjectReaderTest.java

```diff
@@ -805,4 +808,111 @@ public int hashCode() {
             return _delegate.hashCode();
         }
     }
+
+
+    public void testReadValueFromFile() throws IOException {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFile.json",
+            a2q("{ 'name': 'John Doe'}"));
+
+        FilePerson bean = newJsonMapper().readerFor(FilePerson.class).readValue(file);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromFile2() throws IOException {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFile2.json",
+            a2q("{ 'name': 'John Doe'}"));
+
+        FilePerson bean = newJsonMapper().reader().readValue(file, FilePerson.class);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromFileWithFormatDetection() throws IOException {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFileWithFormatDetection.json",
+            a2q("{ 'name': 'John Doe'}"));
+        ObjectReader reader = newJsonMapper().readerFor(FilePerson.class);
+        reader = reader.withFormatDetection(reader);
+
+        FilePerson bean = reader.readValue(file);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromNonExistentFile() throws IOException {
+        File file = new File("testReadValueFromNonExistentFile.json");
+        try {
+            newJsonMapper().readValue(file, FilePerson.class);
+            fail("should not pass");
+        } catch (FileNotFoundException e) {
+            verifyException(e, "testReadValueFromNonExistentFile");
+        }
+    }
+
+    public void testInputStreamFromEmptyFile() throws IOException {
+        File file = new File("testInputStreamFromEmptyFile.json");
+        assertTrue(file.createNewFile());
```

Comment:
Change to use same functionality for creating empty file -- and not use main-level files.